snuba/web/db_query.py

```diff
@@ -63,11 +63,12 @@ def update_query_metadata_and_stats(
 
 @with_span(op="db")
 def execute_query(
+    # TODO: Passing the whole clickhouse query here is needed as long
+    # as the execute method depends on it. Otherwise we can make this
+    # file rely either entirely on clickhouse query or entirely on
+    # the formatter.
     clickhouse_query: Query,
     request_settings: RequestSettings,
-    # TODO: Passing the SqlQuery (which basically wraps the query and formats it)
```

Comment:
I think this comment was on the wrong method. At this point of the processing I would keep SqlQuery around and stop depending on ClickhouseQuery which is the AST.